source/MaterialXTest/ShaderValid.cpp

```diff
@@ -0,0 +1,291 @@
+    shaderGenerator->registerSourceCodeSearchPath(searchPath);
+
+    // Create a nonsensical graph testing some geometric nodes
+    mx::NodeGraphPtr nodeGraph = doc->addNodeGraph("geometry_attributes");
+
+    std::vector<mx::NodePtr> attributeList;
+
+    // normal output test
+    mx::NodePtr normal1 = nodeGraph->addNode("normal", "normal1", "vector3");
+    normal1->setParameterValue("space", std::string("world"));
+    attributeList.push_back(normal1);
+
+    // position output test
+    mx::NodePtr position1 = nodeGraph->addNode("position", "position1", "vector3");
+    position1->setParameterValue("space", std::string("world"));
+    mx::NodePtr constant = nodeGraph->addNode("constant");
+    mx::ParameterPtr v = constant->addParameter("value", "vector3");
+    v->setValueString("0.003921568627451, 0.003921568627451, 0.003921568627451");
+    mx::NodePtr multiply1 = nodeGraph->addNode("multiply", "multiply1", "vector3");
+    multiply1->setConnectedNode("in1", position1);
+    multiply1->setConnectedNode("in2", constant);
+    //attributeList.push_back(multiply1); -- error : Could not find a nodedef for node 'multiply1' ?
+    attributeList.push_back(position1);
+
+    // color output test
+    mx::NodePtr geomcolor1 = nodeGraph->addNode("geomcolor", "geomcolor1", "color3");
+    geomcolor1->setParameterValue("index", 0, "integer");
+    attributeList.push_back(geomcolor1);
+
+    // tangent output test
+    mx::NodePtr tangent = nodeGraph->addNode("tangent", "tangent1", "vector3");
+    //tangent->setParameterValue("index", 0, "integer"); -- not supported yet in core
+    attributeList.push_back(tangent);
+
+    // bitangent output test
+    mx::NodePtr bitangent = nodeGraph->addNode("bitangent", "bitangent1", "vector3");
+    //bitangent->setParameterValue("index", 0, "integer"); -- not supported yet in core
+    attributeList.push_back(bitangent);
+
+    // uv output test
+    mx::NodePtr texcoord1 = nodeGraph->addNode("texcoord", "texcoord1", "vector2");
+    texcoord1->setParameterValue("index", 0, "integer");
+    mx::NodePtr swizzle1 = nodeGraph->addNode("swizzle", "swizzle_uv", "vector3");
+    swizzle1->setConnectedNode("in", texcoord1);
+    swizzle1->setParameterValue("channels", std::string("xy0"));
+    attributeList.push_back(swizzle1);
+
+    // image
+    mx::FilePath imagePath = mx::FilePath::getCurrentPath() / mx::FilePath("documents/Images/MaterialXLogo.exr");
+    std::string imageName = imagePath.asString();
+    mx::NodePtr image = nodeGraph->addNode("image", "image1", "color3");
+    image->setParameterValue("file", imageName, "filename");
+    attributeList.push_back(image);
+
+    // Connected to output.
+    mx::OutputPtr output1 = nodeGraph->addOutput(mx::EMPTY_STRING, "vector3");
+
+    mx::GlslValidator validator;
+    validator.initialize();
+    REQUIRE(validator.iniitialized());
+
+    mx::TinyEXRImageHandlerPtr handler = std::make_shared<mx::TinyEXRImageHandler>();
```

Comment:
To follow the convention in the MaterialXShaderGen library we could hide the memory allocation using a creator method:
`mx::TinyEXRImageHandlerPtr handler = mx::TinyEXRImageHandler::create();`